src-cpp/rdkafkacpp.h

```diff
@@ -1800,14 +1800,17 @@ class RD_EXPORT Headers {
      *
      * @remark The error code is used for when the Header is constructed
      *         internally by using RdKafka::Headers::get_last which constructs
-     *         a Header encapsulating the ErrorCode in the process
+     *         a Header encapsulating the ErrorCode in the process.
+     *         In case of error, value should not be accessed, it ignores
```

Comment:
`value and value_size will be undefined in the case of error.`

Reply:
Here is a catch, while your C api states above, Your C++ api does not.
For example, your destructor deallocates array undefined or not.
```
     if (value_ != NULL)
        free(value_);
```
While undefined value_size has no consequence access problems, undefined pointer has.
it is possible to do 2 different things.
1) set value_= NULL when error in single place.
2) if (err==ERR_NO_ERROR && value_ != NULL) <do some> else ...

Reply:
Yes, we should always set value_, either to the allocated space or NULL (on error).
But from the user's perspective they should consider value and size undefined in case of error.